### DISCUSSION TITLE
Typescript usage: how do define test?

### DISCUSSION THREAD

**Discussion by dmelo7** [artilleryio/artillery#3341] [Artillery]
Hi everyone I recently discovered Artillery and looks like a really good solution for the project I am working for. We have an extensive playwright testing solution and we want start adding some load testing too. For Artillery to be the best solution we want to be able to reuse our code, so stick with TS and re utilize the page object. 

On the docs, it mentions the usage of `test.step` but on typescript I can not type this correctly. Page comes from playwright but where does test come from? How to type this?

By the way, `test` from artillery and `test` from playwright are not the same right? On artillery processors the only function I can use is `.step`? I was wondering if we could reuse our fixtures too but it's not clear and I failed all the attempts! Can't call the playwright `test` inside the processors.

Thanks all :) 

Edit:
Add doc reference:
https://www.artillery.io/docs/reference/engines/playwright#teststep-argument 

Exemple code:

```
import AppPage from '#page-objects/app/AppPage'
import Login from '#page-objects/app/Login'
import { users } from '#utils/constants'
import { Page } from '@playwright/test'

export async function loginTest(page: Page, test: ????) {
  const db = await Database.new()
  const loginPage = new Login(page)
  const appPage = new AppPage(page)

  await page.goto('http://localhost:3001/')

  await test.step('login', async () => {
    await loginPage.login({ user: users.testUser})
  })
  await test.step('logout', async () => {
    await appPage.menu.logout()
  })
}
```

**Comment by hassy** [comment]
hi @dmelo7 👋 

> Page comes from playwright but where does test come from? 

Yep that's right. The `step` argument in Artillery's hook function is Artillery's version of Playwright's `test.step()` helper. Takes the same first two arguments, defined here if you're curious: https://github.com/artilleryio/artillery/blob/main/packages/artillery-engine-playwright/index.js#L102-L115

> Can't call the playwright test inside the processors.

Yes correct. Artillery's integration is focused on using Playwright as a library, it's not fully compatible with the Playwright Test Runner. You _can_ use functions like `expect` from the `@playwright/test`, but Artillery can't run Playwright Test Runner tests defined with the `test()` function.

**Reply by dmelo7** [reply]
Hey @hassy thanks for clarifying it. I was unsure on what I could do with playwright but now it's clear. I am still not sure on how to type this test using typescript, it is not recognisable by the vscode too. Any suggestion? Thanks!

**Comment by dmelo7** [comment]
Any suggestions?

**Comment by dmelo7** [comment]
I still can't find how to properly type these parameters: vuContext, events, test.

**Reply by Anonymvs17** [reply]
@dmelo7 I have not found the type but for me at least it is running with: _(page: Page, vuContext: any, events: any, test: any)_. So I assigned the type **any**. In this case it is also important to have vuContext & events in there even if you do not use them.
